### PR TITLE
Rev-022-3: create features selection component

### PR DESCRIPTION
**Pull request by Anas-AlRamahi**
### Description
Added the feature selection section in create new listing page.

Links:
[Reverifi Figma design](https://www.figma.com/file/22d3bKcFL0ipWIZJOUY3Km/reverifi?node-id=0%3A1)
[Reverifi Ticket Excel sheet](https://docs.google.com/spreadsheets/d/1o7cWWsGAgbzCfvuCwblf2ExczTyPkWbfx2UhRbEpRLo/edit#gid=0)

### CODE REVIEW

File: src/components/FeatureSelection/feature-selection.styles.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  align-items: center;
+  color: ${(props) => (props.active ? colors.green : colors.gray)};
+  display: flex;
+  font-size: 1rem;
+  font-weight: 600;
+  user-select: none;
+
+  :hover {
+    cursor: pointer;
+  }
```
Comment (mohdabed123) [inline src/components/FeatureSelection/feature-selection.styles.js:49]:
```suggestion
    cursor: pointer;
```

File: src/components/FeatureSelection/feature-selection.styles.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    margin: 0 0 2rem 0;
+  `}
+`;
+
+export const FeatureTitle = styled.label`
+  align-items: center;
+  color: ${(props) => (props.active ? colors.green : colors.gray)};
+  display: flex;
+  font-size: 1rem;
+  font-weight: 600;
```
Comment (mohdabed123) [inline src/components/FeatureSelection/feature-selection.styles.js:44]:
```suggestion
  font-weight: bold;
```

File: src/components/FeatureSelection/index.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  /**
+   * Click handling function that will add or remove the feature id from the array.
+   *
+   * @param {event} event The event triggered by clicking on a feature element.
+   * @param {number} id Id of the event invoking element.
+   */
+  const handleClick = (event, id) => {
+    event.preventDefault();
+    if (featureIds.has(id)) {
+      featureIds.delete(id);
```
Comment (mohdabed123) [inline src/components/FeatureSelection/index.jsx:24]: can we not use state directly maybe use previous state

File: src/components/FeatureSelection/index.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import data from './data';
+import {
+  Feature,
+  FeaturesContainer,
+  FeatureTitle,
+  IconContainer,
+  MainContainer,
+  Title,
+} from './feature-selection.styles';
+
```
Comment (mohdabed123) [inline src/components/FeatureSelection/index.jsx:11]: docs here

File: src/components/FeatureSelection/feature-selection.styles.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    props.active ? `0 0 0.62rem ${colors.green}` : `none`};
+  display: flex;
+  justify-content: center;
+  margin-right: 0.25rem;
+  min-height: 2rem;
+  min-width: 2rem;
+
+  img {
+    opacity: ${(props) => (props.active ? 1 : 0.5)};
+  }
```
Comment (mohdabed123) [inline src/components/FeatureSelection/feature-selection.styles.js:66]:
```suggestion
background: ${(props) => (props.active ? colors.green : colors.white)};
  border-radius: 50%;
  box-shadow: ${(props) =>
    props.active ? `0 0 0.62rem ${colors.green}` : `none`};
  display: flex;
  margin-right: 0.25rem;
  min-height: 2rem;
  min-width: 2rem;

  img {
    opacity: ${(props) => (props.active ? 1 : 0.5)};
  }
```

**Review by mohdabed123 — COMMENTED** [review]
Note: there is a similar component used in this [PR](https://github.com/abukhadraop-or/reverifi/pull/20) that is read only , we might want to create a ticket to clean up one of them and make the other re-usable so it can be used as read only or a select in the create form

File: src/components/FeatureSelection/feature-selection.styles.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+export const MainContainer = styled.div`
+  padding: 2rem 1rem;
+
+  ${mq.mobile`
+    padding: 2rem 3.75rem;
+  `}
+`;
+
+export const Title = styled.h3`
+  margin: 0 0 1.43rem 0.12rem;
```
Comment (AhmadAlShwaiki) [inline src/components/FeatureSelection/feature-selection.styles.js:14]: Do we really need these arbitrary values?
```suggestion
```

**Review by AhmadAlShwaiki — APPROVED** [review]
Minor changes and suggestions, minor so approving early.

**Review by mohdabed123 — APPROVED** [review]

**Review by AhmadAlShwaiki — APPROVED** [review]

**Review by AhmadAlShwaiki — APPROVED** [review]
Looks good

**Pull request merged**